lualib/resty/kong/peer_conn.lua

```diff
@@ -6,16 +6,17 @@ local errmsg           = base.get_errmsg_ptr()
 local C                = ffi.C
 local ffi_str          = ffi.string
 local get_phase        = ngx.get_phase
+local subsystem        = ngx.config.subsystem
 local NGX_ERROR        = ngx.ERROR
 
 local error            = error
 
-
-ffi.cdef[[
-int
-ngx_http_lua_kong_ffi_get_last_peer_connection_cached(ngx_http_request_t *r,
-    char **err);
-]]
+if subsystem == "http" then
```

Comment:
seemly it is unnecessary. I guess it always cdef this function in our scenario? Or some stream part lua code will require this lib?

Reply:
stream part lua code doesn't need this lib, but I found it would be search and failed in stream subsystem,

```
Error: ./kong/cmd/start.lua:99: nginx: [error] init_by_lua error: ...build/kong-dev/openresty/lualib/resty/kong/peer_conn.lua:18: declaration specifier expected near 'ngx_http_request_t' at line 2
stack traceback:
	[C]: in function 'cdef'
	...build/kong-dev/openresty/lualib/resty/kong/peer_conn.lua:18: in main chunk
	[C]: in function 'require'
	./kong/init.lua:98: in main chunk
	[C]: in function 'require'
	init_by_lua:22: in main chunk
```

ref: https://github.com/Kong/kong/actions/runs/8353645655/job/22865711138?pr=12760

Reply:
cool